src/backend/columnar/cstore_reader.c

```diff
@@ -127,6 +137,23 @@ ColumnarReadNextRow(TableReadState *readState, Datum *columnValues, bool *column
 	StripeMetadata *stripeMetadata = readState->currentStripeMetadata;
 	MemoryContext oldContext = NULL;
 
+	/*
+	 * We rely on first column's metadata in rest of this function. So for
+	 * zero column tables we just return "true" for totalRowCount times.
+	 */
+	if (readState->tupleDescriptor->natts == 0)
```

Comment:
I changed this to `readstate->projectedColumnList == NIL` & added a line to memset `columnNulls` to true. In case of zero-columns with deleted attributes we still have `columnar.chunk` metadata and we didn't have to handle them here.

Reply:
Also we have a regression test for such a case.